### PR TITLE
Converted isTemporal, nestedDeadlock, isReachability, isLoopSensitive, containsNext to visitor

### PR DESCRIPTION
**Pull request by Ragusaen**

### CODE REVIEW

File: include/PetriEngine/PQL/Visitor.h [hunk 1]
```diff
@@ -368,116 +329,17 @@ namespace PetriEngine {
...
-                (*condition)[0]->visit(*this);
+            void _accept(const SimpleQuantifierCondition *condition) override {
+                condition->getCond()->visit(*this);
             }
 
             void _accept(const UntilCondition *condition) override
             {
-                (*condition)[0]->visit(*this);
-                (*condition)[1]->visit(*this);
+                condition->getCond1()->visit(*this);
```
Comment (petergjoel) [inline include/PetriEngine/PQL/Visitor.h:338]: Keep the old `[]` notation instead, it would be nice to cut down on the number of specialized getters - in particular when they carry no significant information like `getCond1()`

File: include/PetriEngine/PQL/XMLPrinter.h [hunk 2]
```diff
@@ -28,16 +28,17 @@ namespace PetriEngine {
...
+                single_tab = std::string(tab_size, ' ');
+            }
 
         protected:
             std::ostream& os;
             const bool token_count;
             uint32_t tabs;
-            const uint32_t tab_size;
             const bool print_newlines;
-            static constexpr char single_tab = ' ';
```
Comment (petergjoel) [inline include/PetriEngine/PQL/XMLPrinter.h:40]: Why is this changed back into a string?

File: src/PetriEngine/PQL/XMLPrinter.cpp [hunk 3]
```diff
@@ -28,14 +28,14 @@ namespace PetriEngine {
...
-            generateTabs() << "<" << tag;
+            generateTabs() << "<" << tag << ">";
             newline();
             tabs++;
         }
 
         void XMLPrinter::closeXmlTag(const std::string &tag) {
             tabs--;
-            generateTabs() << "</" << tag;
+            generateTabs() << "</" << tag << ">";
```
Comment (petergjoel) [inline src/PetriEngine/PQL/XMLPrinter.cpp:38]: Another time, make such a fix a separate PR.

**Review by petergjoel — CHANGES_REQUESTED** [review]
A few small things to fix, otherwise looks good. I will run some tests (where we expect to see a slowdown due to the temporal check not being catched).

File: include/PetriEngine/PQL/XMLPrinter.h (hunk 2 above)
Comment (Ragusaen) [reply include/PetriEngine/PQL/XMLPrinter.h:40]: The tab size is set by `tab_size` in the constructor, so a single tab can be more than one space. Before the tab_size was unused. It is more convenient to write `<< single_tab << "something else"` compared to something like `<< repeat(single_tab, tab_size) << "something else"`.

**Comment by petergjoel** [comment]
Interestingly this has no (negative) impact on the performance of CTL for colored nets. 

I am awaiting cluster-time for a check on regular nets, but otherwise, it looks good to me.

**Review by petergjoel — APPROVED** [review]
Checks out on cluster. 
@srba  please merge.
